File: ruby_solutions/two_sum.rb
# frozen_string_literal: true

def two_sum(nums, target)
  nums.each_with_index.each_with_object({}) do |(number, index), map|
    difference = target - number
    return [map[difference], index] if map.key?(difference)

    map[number] = index
  end
end

p two_sum([3, 3], 6)

File: js_solutions/two-sum.js
function twoSum(nums, target) {
  const map = {};
  for (let i = 0; i < nums.length; i++) {
    const difference = target - nums[i];
    if (difference in map) return [map[difference], i];

    map[nums[i]] = i;
  }
}

console.log(twoSum([3, 3], 6)); // [0, 1]
